tests/unittest.py

```diff
@@ -155,20 +155,78 @@ def run_to_completion(self):
             self.clock.advance(amount)
 
     def assertTipsEqual(self, manager1, manager2):
-        s1 = set(manager1.tx_storage.get_all_tips())
-        s2 = set(manager2.tx_storage.get_all_tips())
+        assert (
+            hasattr(self, '_enable_sync_v1') and
+            hasattr(self, '_enable_sync_v2') and
+            (self._enable_sync_v1 or self._enable_sync_v2)
+        ), (
+            'Please set both `_enable_sync_v1` and `_enable_sync_v2` on the class. '
+            'Also they can\'t both be False. '
+            'This is by design so it\'s we don\'t forget to test for multiple sync versions.'
+        )
+        if self._enable_sync_v2:
+            self.assertTipsEqualSyncV2(manager1, manager2)
+        else:
+            self.assertTipsEqualSyncV1(manager1, manager2)
+
+    def assertTipsEqualSyncV1(self, manager1, manager2):
+        # tx tips
+        self.assertEqual(manager1.tx_storage._mempool_tips_index, manager2.tx_storage._mempool_tips_index)
+
+        # best block
+        s1 = set(manager1.tx_storage.get_best_block_tips())
```

Comment:
Maybe use `get_all_tips()`? Best block tips do not cover all tips because we also have the mempool tips.

Reply:
I just did this change here: 95a8188f3960e92a51f4c9e4d5d9d7df851c01ee
@jansegre

Reply:
Old conversation. It does make sense anymore. Marking as resolved.